crates/tiny/src/main.rs

```diff
@@ -104,25 +104,26 @@ fn run(
     local.block_on(&runtime, async move {
         // Create TUI task
         let (tui, rcv_tui_ev) = TUI::run(config_path.clone());
-        tui.draw();
 
         // Create logger
         let report_logger_error = {
             let tui_clone = tui.clone();
             Box::new(move |err: String| {
                 // Somehwat hacky -- only tab we have is "mentions" so we show the error there
```

Comment:
This comment can be removed, since now an "errors" tab should pop up, but...

@osa1 do you know how to force this state so I can test it?

Reply:
@trevarj you need to somehow return the error value here: https://github.com/osa1/tiny/blob/be66a8026314bbcce98d939401a79c42d279dc56/crates/libtiny_logger/src/lib.rs#L140-L147

Not sure how to add a unit test for this, but for development purposes I think you can modify that code to return an error always.

Reply:
@osa1 what are your thoughts on the UX of this? The "errors" tab will only pop up if there is an issue with the logger (ex. setting a bad path). If you close it then `tiny` shuts down. We can make it so that it can't be closed, then the user will need to fix the error to make it go away.

My opinion is to name the tab "logger" and then also make it not able to close (with /close)

Reply:
From UX point of view I think it should be possible to close the "errors" tab.

Here's an idea that I think would work great (though it may be difficult to implement): when all tabs closed (or when no tabs are crated when you run tiny) we will show the "errors" tab, with an input line (so that `/connect` etc. will work), but only if there is at least one error. If the user closes it, then it won't appear again until there's a new error. When it appears, it will appear as the first tab.

Also, if there is only the "errors" tab in the TUI then we may not want to show it in the tab bar.

More concretely:

- If I run the tiny but no server tabs are created, this is what I will see:
  ```
  |                        | // messaging area
  |█                       | // input line
  |                        | // tabs
  ```
  Note that tab line is empty even though we're in `errors` tab

- In case of an error:
  ```
  |Can't create logger: ...| // messaging area
  |█                       | // input line
  |                        | // tabs
  ```
  Ditto, tab line is empty

- When a new tab is created:
  ```
  |Can't create logger: ...| // messaging area
  |█                       | // input line
  |errors irc.oftc.net     | // tabs
  ```
  Note that tab line is empty

- After `/close` in `errors` tab:
  ```
  |...                     | // messaging area
  |█                       | // input line
  |irc.oftc.net            | // tabs
  ```

- After `/close` in `irc.oftc.net` tiny we exit.

@trevarj What do you think about this design in terms of UX and implementation complexity?

Reply:
I like the idea of the blank tab, but it's definitely more changes to add a special case for hiding the tab in the tab line, I think. Might not be worth the mess, but I can look into it.


What would be *really* cool is if we could somehow make it like a welcome screen like vim/emacs. Maybe have it list some of the main commands (/connect, /close, etc). 

```
          ______    _                  
         /_  __/   (_)   ____    __  __
          / /     / /   / __ \  / / / /
         / /     / /   / / / / / /_/ / 
        /_/     /_/   /_/ /_/  \__, /  
                              /____/ IRC
             Basic commands:
             /connect - blah blah
             /join - asdasd
             /close - closes tab

Error: something went wrong...
█ 
welcome irc.oftc.net
```